### ISSUE TITLE
Encoding::UndefinedConversionError in polling mode when values contain invalid UTF-8 sequences

### ISSUE THREAD

**Issue by Manfred** [launchdarkly/ruby-client#90]
When the client is running in polling mode and any of the HTTP responses from Launch Darkly contain invalid UTF-8 characters it will throw an exception and return the default value for the variation. In our specific case it's a custom key containing a customer name with "\xE2\x80\x99" (apostrophe) in it somewhere.

Apparently this is a Faraday HTTP cache problem and it was previously mentioned in a PR:
 https://github.com/launchdarkly/ruby-client/pull/51#issuecomment-178281150

Our current workaround is to set the serializer to Marshal:

```
require 'ldclient-rb/requestor'

module LaunchDarkly
  class Requestor
    def initialize(sdk_key, config)
      @sdk_key = sdk_key
      @config = config
      @client = Faraday.new do |builder|
        builder.use :http_cache, store: @config.cache_store, serializer: Marshal
        builder.adapter :net_http_persistent
      end
    end
  end
end
```

This problem exists in `2.2.7`.

**Comment by eli-darkly** [comment]
Sorry this fell through the cracks. I'm guessing that we haven't heard about it since then because most people are now using streaming mode, but I verified that we do still have this bug in polling mode. The proposed fix does seem effective, but we're still looking at whether it's the right way to go. But in any case we will be rolling out some fix as soon as possible.

**Comment by eli-darkly** [comment]
Finally fixed this in version 5.5.0.
